qcodes/dataset/param_spec.py

```diff
@@ -1,19 +1,22 @@
-from typing import Union, Sequence, List
+from typing import Union, Sequence, List, Dict, Any
+from copy import deepcopy
+
 
 # TODO: we should validate type somehow
 # we can't accept everything (or we can but crash at runtime?)
 # we only support the types in VALUES type, that is:
 # str, Number, List, ndarray, bool
 class ParamSpec:
 
+    version = 0  # used when serializing
```

Comment:
Is versioning of  `ParamSpec` enough? What happens when `RunDescription` changes, what version shall be changed then? I'd suggest to version only the top object (in our case, `RunDescription`) so that there is only one version that we want to care about.